src/java.base/share/classes/java/util/HexFormat.java

```diff
@@ -403,21 +407,20 @@ public <A extends Appendable> A formatHex(A out, byte[] bytes, int fromIndex, in
         int length = toIndex - fromIndex;
         if (length > 0) {
             try {
-                out.append(prefix);
-                toHexDigits(out, bytes[fromIndex]);
-                if (suffix.isEmpty() && delimiter.isEmpty() && prefix.isEmpty()) {
-                    for (int i = 1; i < length; i++) {
-                        toHexDigits(out, bytes[fromIndex + i]);
-                    }
+                String s = formatOptDelimiter(bytes, fromIndex, toIndex);
+                if (s != null) {
+                    out.append(s);
                 } else {
+                    out.append(prefix);
+                    toHexDigits(out, bytes[fromIndex]);
                     for (int i = 1; i < length; i++) {
                         out.append(suffix);
                         out.append(delimiter);
                         out.append(prefix);
                         toHexDigits(out, bytes[fromIndex + i]);
                     }
+                    out.append(suffix);
```

Comment:
Maybe change this whole `else` block to
```java
                    for (int i = 0; i < length; i++) {
                        if (i > 0)
                            out.append(delimiter);
                        out.append(prefix);
                        toHexDigits(out, bytes[fromIndex + i]);
                        out.append(suffix);
                    }
```
for clarity?

Reply:
The original (and current) is coded to avoid a condition inside the loop.

Reply:
I also think that the way of writing for_0 combined with if > 0 is easier to understand, The operation overhead of if > 0 is very small, and it will not affect performance when used in a loop.